src/cascadia/TerminalCore/Terminal.hpp

```diff
@@ -149,6 +149,27 @@ class Microsoft::Terminal::Core::Terminal final :
     void SetEndSelectionPosition(const COORD position);
     void SetBoxSelection(const bool isEnabled) noexcept;
 
+    enum class Direction
+    {
+        Left,
+        Right,
+        Up,
+        Down
+    };
+    enum class SelectionExpansionMode

```

Comment:
Shouldn't these three enums just be the ones defined in `Terminal.Settings`? For ex, this `SelectionExpansionMode` has a `Line`, while the other doesn't...

Reply:
Talked to @DHowett-MSFT. We should be keeping Terminal.Settings outside of TerminalCore. So, this one won't be doable. I'll have to translate it at the TerminalControl layer.

Reply:
Woah that's straight up not true - `TerminalSettings` is _supposed_ to be lower than `Core` - there's even a `ICoreSettings` that the `Terminal` core uses.

In an ideal world, sure, TerminalCore would expose the ICoreSettings interface, and `TerminalSettings` would live on top of `TerminalCore`, but that's not how it is currently

Reply:
I think in an ideal world, TerminalCore would _consume_ the ICoreSettings interface, right? It's not intended to expose 50 setters off the Terminal, it's intended for the terminal to use 50 _getters_ (since the ICoreSettings could be implemented by TerminalApp::Profile f.example)